### PR TITLE
rails 4.0 support

### PR DESCRIPTION
**Pull request by grosser**
@pschambacher @eac 

now also supporting procs and detecting their preloadability based on if they get arguments

### CODE REVIEW

File: test/helper.rb [hunk 1]
```diff
@@ -16,40 +18,9 @@
 require_relative 'models'
 
 def assert_queries(num = 1)
-  ActiveRecord::SQLCounter.log = []
-  yield
-ensure
-  assert_equal num, ActiveRecord::SQLCounter.log.size, "#{ActiveRecord::SQLCounter.log.size} instead of #{num} queries were executed.#{ActiveRecord::SQLCounter.log.size == 0 ? '' : "\nQueries:\n#{ActiveRecord::SQLCounter.log.join("\n")}"}"
```
Comment (grosser) [inline test/helper.rb]: never do assertions in an ensure: when something blows up, the assertion hides it by blowing up itself :)

File: test/helper.rb [hunk 2]
```diff
@@ -16,40 +18,9 @@
...
-end
-
-# Yanked from ActiveRecord tests
-module ActiveRecord
```
Comment (grosser) [inline test/helper.rb]: replacing this with query diet since it was not working very well

**Comment by grosser** [comment]
@staugaard

**Comment by pschambacher** [comment]
LGTM :+1:

**Pull request merged**